docs/defs.md

```diff
@@ -0,0 +1,72 @@
+<!-- Generated with Stardoc: http://skydoc.bazel.build -->
+
+Starlark rules for building protocol buffers.
+
+<a id="proto_descriptor_set"></a>
+
+## proto_descriptor_set
+
+<pre>
+proto_descriptor_set(<a href="#proto_descriptor_set-name">name</a>, <a href="#proto_descriptor_set-deps">deps</a>)
+</pre>
+
+Collects all `FileDescriptorSet`s from `deps` and combines them into a single
+`FileDescriptorSet` containing all the `FileDescriptorProto`.
+
+**ATTRIBUTES**
+
+
+| Name  | Description | Type | Mandatory | Default |
+| :------------- | :------------- | :------------- | :------------- | :------------- |
+| <a id="proto_descriptor_set-name"></a>name |  A unique name for this target.   | <a href="https://bazel.build/concepts/labels#target-names">Name</a> | required |  |
+| <a id="proto_descriptor_set-deps"></a>deps |  Sequence of <code>ProtoInfo</code>s to collect <code>FileDescriptorSet</code>s from.   | <a href="https://bazel.build/concepts/labels">List of labels</a> | optional | <code>[]</code> |
+
+
+<a id="proto_lang_toolchain"></a>
+
+## proto_lang_toolchain
+
+<pre>
+proto_lang_toolchain(<a href="#proto_lang_toolchain-name">name</a>, <a href="#proto_lang_toolchain-toolchain_type">toolchain_type</a>, <a href="#proto_lang_toolchain-exec_compatible_with">exec_compatible_with</a>, <a href="#proto_lang_toolchain-target_compatible_with">target_compatible_with</a>, <a href="#proto_lang_toolchain-attrs">attrs</a>)
+</pre>
+
+Creates a proto_lang_toolchain and corresponding toolchain target.
+
+Toolchain target is only created when toolchain_type is set.
+
+https://docs.bazel.build/versions/master/be/protocol-buffer.html#proto_lang_toolchain
+
+
+**PARAMETERS**
+
+
+| Name  | Description | Default Value |
+| :------------- | :------------- | :------------- |
+| <a id="proto_lang_toolchain-name"></a>name |  name of the toolchain   |  none |
+| <a id="proto_lang_toolchain-toolchain_type"></a>toolchain_type |  The toolchain type   |  <code>None</code> |
+| <a id="proto_lang_toolchain-exec_compatible_with"></a>exec_compatible_with |  ([constraints]) List of constraints the prebuild binaries is compatible with.   |  <code>[]</code> |
+| <a id="proto_lang_toolchain-target_compatible_with"></a>target_compatible_with |  ([constraints]) List of constraints the target libraries are compatible with.   |  <code>[]</code> |
+| <a id="proto_lang_toolchain-attrs"></a>attrs |  Rule attributes   |  none |
+
+
+<a id="proto_library"></a>
+
+## proto_library
```

Comment:
proto_toolchain rules is missing in the docs

Reply:
https://github.com/bazelbuild/rules_proto/blob/main/proto/README.md according to this doc, only public API is defs.bzl and repositories.bzl. is that still valid? if so we should fix that doc

Reply:
I think the solution is to add proto_toolchain to defs.bzl. It's on the same level as `proto_lang_toolchain`. So lets document both.

Reply:
To answer your question about https://github.com/bazelbuild/rules_proto/blob/main/proto/README.md
- Kind a everything that's not in /private is public (I think buldifier has a check for that) 
- In my opinion you can delete README.md (any user of rules_proto shouldn't be expected to read and follow it; users should just need to follow general guidelines, and those are not to use things in /private)
- What I think about defs.bzl is that it's legacy, see https://docs.google.com/document/d/1L1JFgjpZ7SrBinb24DC_5nTIELeYDacikcme-YcA7xs/edit#heading=h.cy0m49ne5key

Reply:
done